### PR TITLE
Task status hover tooltip

### PR DESCRIPTION
**Pull request by rkomiyama**
## Description of the PR
Added functionality for tooltip when hovering over the task status icon.

### Issues Related
#216

### CODE REVIEW

File: test/task_card.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const Application = require('spectron').Application;
 const assert = require('assert');
 
-describe('Application launch', function () {
+describe('Spectron tests', function () {
```
Comment (angrykoala) [inline test/task_card.test.js]: This shouldn't have the name of the framework we are using, but describe what are we testing

File: app/renderer/components/tooltip.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    template: `
+    <div class="tooltip">
+        <div class="tooltip-text">
+            {{ status }}
+        </div>
+    </div>
+    `,
+    computed: {
+        status() {
+            switch (this.color) {
```
Comment (angrykoala) [inline app/renderer/components/tooltip.js]: We shouldn't be using a prop called "color" for something that is not defining a color.

For this, we should be using the enumeration defined app/common/task_status
For example:
```js
case TaskStatus.idle:
     return "Idle" 
```

Instead of the prop "color" we should send "status" to the tooltip component

File: app/renderer/components/tooltip.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                    return "Stopped";
+                case "red":
+                    return "Error";
+                case "blue":
+                    return "Running";
+                case "green":
+                    return "OK";
+                case "grey":
+                default:
+                    return "N/A";
```
Comment (angrykoala) [inline app/renderer/components/tooltip.js]: In the default case, the tooltip shouldn't be shown. There is no grey color afaik

File: resources/styles.css [hunk 4]
```diff
@@ -73,12 +106,6 @@
     cursor: default;
 }
 
-.open a.edit-button.tap-target-origin, .open a.edit-button.tap-target-origin:hover {
```
Comment (angrykoala) [inline resources/styles.css]: Why is this removed? I assume is old code, but I'm not sure what was this for

File: resources/styles.css [hunk 5]
```diff
@@ -147,7 +174,6 @@ input.tab-text {
 
 .unselectable-text {
     -webkit-user-select: none;
-    cursor: default;
```
Comment (angrykoala) [inline resources/styles.css]: why is this removed?

File: resources/styles.css [hunk 6]
```diff
@@ -250,7 +280,6 @@ main {
     bottom: 56px;
     left: 0;
     right: 0;
-    overflow: auto;
```
Comment (angrykoala) [inline resources/styles.css]: Removing this breaks the behavour of the bottom bar.

Please, make sure you properly test any change made to the css

File: test/task_card.test.js [hunk 7]
```diff
@@ -23,4 +23,19 @@ describe('Application launch', function () {
             assert.equal(count, 1);
         });
     });
+
+    it('shows tooltip on mouse hover', function () {
+        let selector = '.task-card:nth-child(1) .tooltip';
```
Comment (angrykoala) [inline test/task_card.test.js]: this should be a const

File: test/task_card.test.js [hunk 8]
```diff
@@ -23,4 +23,19 @@ describe('Application launch', function () {
             assert.equal(count, 1);
         });
     });
+
+    it('shows tooltip on mouse hover', function () {
```
Comment (angrykoala) [inline test/task_card.test.js]: please, use arrow function ( `()=>{`) for anonymous function

File: test/task_card.test.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 const Application = require('spectron').Application;
 const assert = require('assert');
 
-describe('Application launch', function () {
+describe('Spectron tests', function () {
     this.timeout(10000);
 
     beforeEach(function () {
```
Comment (angrykoala) [inline test/task_card.test.js]: Use `()=>{` here as well

File: test/task_card.test.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
 const Application = require('spectron').Application;
 const assert = require('assert');
 
-describe('Application launch', function () {
+describe('Spectron tests', function () {
     this.timeout(10000);
```
Comment (angrykoala) [inline test/task_card.test.js]: Is this timeout required?

File: resources/styles.css [hunk 11]
```diff
@@ -53,6 +53,39 @@
...
+  content: '';
+  position: absolute;
+  top: 100%;
+  right: 10px;
+  border-width: 5px;
+  border-style: solid;
+  border-color: black transparent transparent transparent;
+}
+
+.tooltip:hover .tooltip-text {
```
Comment (angrykoala) [inline resources/styles.css]: Have you tried using a tooltip defined in materialize to improve the behavior of opening and closing? (some kind of animation or something?)

File: resources/styles.css (hunk 4 above)
Comment (rkomiyama) [reply resources/styles.css]: Not sure. Must have happened when I merged or something.

File: resources/styles.css (hunk 6 above)
Comment (rkomiyama) [reply resources/styles.css]: screen shot 2017-11-01 at 10 32 10 am

If `main` is set to `overflow: auto`, the top bar overlaps the tooltip.

File: test/task_card.test.js (hunk 10 above)
Comment (rkomiyama) [reply test/task_card.test.js]: I don't think so. Removing it should be okay.

**Comment by rkomiyama** [comment]
@angrykoala I've addressed your comments, but we still need to fix the tooltip overlap issue.

File: test/task_card.test.js (hunk 10 above)
Comment (rkomiyama) [reply test/task_card.test.js]: Actually, this might be necessary for Travis.

File: test/task_card.test.js (hunk 8 above)
Comment (rkomiyama) [reply test/task_card.test.js]: I can do that, but Mocha throws an error when I try to use arrow function for `describe()`.

Comment (angrykoala) [reply test/task_card.test.js]: Every other `describe` block in gaucho is using arrow function, so I don't understand why mocha would fail

File: app/renderer/components/task_card.js [hunk 12]
```diff
@@ -39,17 +41,18 @@ module.exports = {
             <div class="col s1">
                 <progress-spinner v-if="running && AppStatus.config.animatedSpinner"></progress-spinner>
                 <i v-else class="small material-icons" v-bind:style="{color: statusColor}">{{task.status}}</i>
+                <tooltip v-bind:task="task"></tooltip>
```
Comment (angrykoala) [inline app/renderer/components/task_card.js]: This is better, but probably binding task.status will make the code in Tooltip easier and less coupled with Task

File: app/renderer/components/tooltip.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+            {{ status }}
+        </div>
+    </div>
+    `,
+    computed: {
+        status() {
+            switch (this.task.status) {
+                case TaskStatus.idle:
+                case TaskStatus.stopped:
+                    return "Stopped";
```
Comment (angrykoala) [inline app/renderer/components/tooltip.js]: Wich is better as text, `stopped` or `idle`? (open to discussion)

File: app/renderer/components/tooltip.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+                case TaskStatus.error:
+                    return "Error";
+                case TaskStatus.running:
+                    return "Running";
+                case TaskStatus.ok:
+                    return "OK";
+                default:
+                    return this.task.status;
```
Comment (angrykoala) [inline app/renderer/components/tooltip.js]: good enough

File: resources/styles.css (hunk 4 above)
Comment (angrykoala) [reply resources/styles.css]: please check what is this for

**Comment by angrykoala** [comment]
screenshot from 2017-11-01 23-40-27
As an example, your changes made the scroll bar always visible

File: app/renderer/components/tooltip.js (hunk 13 above)
Comment (rkomiyama) [reply app/renderer/components/tooltip.js]: I feel like `stopped` is better since `idle` makes it sound like it's suspended.

File: resources/styles.css (hunk 4 above)
Comment (rkomiyama) [reply resources/styles.css]: I really have no idea, so I'll just put it back in.

**Comment by rkomiyama** [comment]
Please let me know exactly which bugs exists. Some of them do not show up on my machine.

**Comment by angrykoala** [comment]
Those where bugs related to the changes in the css you made. your last changes fixed them for what I can see.

I'll check the css carefully and let you know, overall the code looks good
Thanks
